### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ] In this project, I had used migration files to build my schema.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  create_table "comments", force: :cascade do |t|
+    t.bigint "AuthorId", null: false
+    t.bigint "post_id", null: false
+    t.string "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_comments_on_AuthorId"
+    t.index ["post_id"], name: "index_comments_on_post_id"
+  end
```
Comment (julie-ify) [inline db/schema.rb]: - [ ] Kindly note that in Rails we use snake_casing for naming columns. Kindly rename `AuthorId` to `author_id` in this comments table.

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  create_table "likes", force: :cascade do |t|
+    t.bigint "AuthorId", null: false
+    t.bigint "post_id", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_likes_on_AuthorId"
+    t.index ["post_id"], name: "index_likes_on_post_id"
+  end
```
Comment (julie-ify) [inline db/schema.rb]: - [ ] Kindly note that in Rails we use snake_casing for naming columns. Kindly rename `AuthorId` to `author_id` in this likes table.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  create_table "posts", force: :cascade do |t|
+    t.bigint "AuthorId", null: false
+    t.string "title"
+    t.string "text"
+    t.integer "comments_count"
+    t.integer "likes_count"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_posts_on_AuthorId"
+  end
```
Comment (julie-ify) [inline db/schema.rb]: - [ ] Kindly note that in Rails we use snake_casing for naming columns. Kindly rename `AuthorId` to `author_id` in this posts table.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Ibtisam34,

Good job so far!

## Highlights!
- Created migrations ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2023_03_07_000000) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.bigint "Author_Id", null: false
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    t.bigint "post_id", null: false
+    t.string "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_comments_on_AuthorId"
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    t.index ["post_id"], name: "index_comments_on_post_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.bigint "Author_Id", null: false
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    t.bigint "post_id", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_likes_on_AuthorId"
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

File: db/schema.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    t.index ["post_id"], name: "index_likes_on_post_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.bigint "Author_Id", null: false
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

File: db/schema.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    t.string "title"
+    t.string "text"
+    t.integer "comments_count"
+    t.integer "likes_count"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["AuthorId"], name: "index_posts_on_AuthorId"
```
Comment (Ghiftee) [inline db/schema.rb]: - [ ] Good job making the corrections, however, when using ```snake_case``` all letters need to be in lowercase, you can read more on [snake casing here](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/). Kindly change ```Author_Id``` to ```author_id```.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Ibtisam34 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ App works as expected
✔️ README is professional

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: db/schema.rb (hunk 7 above)
Comment (Ibtisam34) [reply db/schema.rb]: https://github.com/Ghiftee hello do I need to chnage this sectin author_id too

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🍾🎉**
Hello @Ibtisam34 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.
- Your linters are passing all the checks.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**